.github/workflows/snap.yaml

```diff
@@ -8,18 +8,23 @@ on:
       - noetic-devel
   workflow_dispatch:
   workflow_call:
+   inputs:
+      branch-name:
+        required: false
+        type: string
+        default: ''
 
 jobs:
   snap:
-    uses: ubuntu-robotics/snap_workflows/.github/workflows/snap.yaml@main
+    uses: ubuntu-robotics/snap_workflows/.github/workflows/snap.yaml@fix/default_branch_name
     secrets:
       store-login: ${{ secrets.STORE_LOGIN }}
     with:
       ubuntu-image: ubuntu-20.04
-      branch-name: noetic-devel
+      branch-name: ${{ inputs.branch-name == '' && github.ref || inputs.branch-name }}
```

Comment:
For future reference, this is a ternary operator-like hack. See e.g. the related [discussion](https://github.com/orgs/community/discussions/75928).